c10/core/AllocatorConfig.cpp

```diff
@@ -0,0 +1,438 @@
+#include <c10/core/AllocatorConfig.h>
+#include <c10/core/DeviceType.h>
+#include <c10/util/CallOnce.h>
+#include <c10/util/env.h>
+
+#include <array>
+#include <cstdlib>
+
+namespace c10::CachingAllocator {
+
+namespace {
+constexpr size_t kRoundUpPowerOfTwoIntervals = 16;
+constexpr size_t kMB = 1024 * 1024ul;
+constexpr size_t kRoundUpPowerOfTwoStart = 1 * kMB; // 1MB
+constexpr size_t kRoundUpPowerOfTwoEnd = 64 * 1024ul * kMB; // 64GB
+constexpr size_t kPinnedMaxRegisterThreads = 128;
+} // anonymous namespace
+
+AllocatorConfig& AllocatorConfig::instance() {
+  static AllocatorConfig instance;
+  static c10::once_flag init_once;
+#define C10_ALLOCATOR_CONFIG_PARSE_ENV(env, deprecated)                       \
```

Comment:
You should either define this outside the function if it used elsewhere/you want it available in the whole file or undef it here.

Reply:
`#undef` added.